### ISSUE TITLE
It should NOT be possible to give sameHgConcept relations to pits with different types

### ISSUE THREAD

**Issue by mmmenno** [erfgoed-en-locatie/histograph#74]
By accident, I related an address to a BAG building id. This works, and I think maybe it shouldn't?

![image](https://cloud.githubusercontent.com/assets/4539618/10942924/6a06dc4a-8312-11e5-9792-a195aa528c60.png)

**Comment by bertspaan** [comment]
Correct. Your data is wrong. Core/Graphmalizer does yet check for such mistakes.

**Comment by wires** [comment]
indeed, shouldn't be possible, but it is more subtle than it seems: you can add a `same` relation between `id`s that are not in the graph yet, so we don't know their types yet...

**Comment by bertspaan** [comment]
We could make system that will disallow those links, such that the respective dataset owners get notified when errors occur.

**Comment by mmmenno** [comment]
Or a system that simply deletes relations like this - let them figure it out themselves!

Not a big priority, as far as i'm concerned.

**Comment by bertspaan** [comment]
We currently also allow somebody added the relation `Utrecht -> hg:sameHgConcept -> Amsterdam`, which would be just as bad for Histograph's integrity.

**Comment by mmmenno** [comment]
But then, I think maybe it should be possible to say a `rm/510640 -> hg:sameHgConcept -> bag/153100000251610`, while the first is a `hg:Monument` and the second a `hg:Building`.

**Comment by sbocconi** [comment]
Is this an issue of (not) allowing sameHgConcept relations between PITs of different types? It reminds me of the the `owl:sameAs` discussion ...
There all is determined by the "semantic strength" of the relation itself (for reasoning), and being sameAs very strong two objects (instances) of different types (classes) cannot be sameAs unless they are also of both types (classes), with all the reasoning consequences of this.

**Comment by mmmenno** [comment]
I think the best solution might be to change the ontology. Now all types are on the same level, making hg:Monument a subclass of hg:Building would solve this issue.

**Comment by tomdemeyer** [comment]
Yes. That sounds like a sensible solution. @bert: what are the implications? I figure not too much?

## 

Tom Demeyer
Waag Society
http://waag.org

**Comment by bertspaan** [comment]
Histograph currently has no notion of subclasses. We could add this, though. I know, however, that @wires was strongly opposed to subtypes. Forgot why. @wires, why?

**Comment by bertspaan** [comment]
And see @wires comment in this issue:

> indeed, shouldn't be possible, but it is more subtle than it seems: you can add a same relation between ids that are not in the graph yet, so we don't know their types yet...

**Comment by mmmenno** [comment]
So, since ...
- you can give a sameAs relation to a non-existing, type-less pit
- sometimes you want a sameAs relation between say a monument and a building
- so far this hasn't been a problem
- you could easily corrupt things by declaring hg:Place Utrecht is sameAs hg:Place Amsterdam as well

... maybe I should just close this issue? Agree?

When this starts to be a problem i gather it won't be difficult to write something to detect and delete sameAs relations between different types (perhaps monument-building excluded)?
